WcaOnRails/app/webpacker/components/Panel/Board/LeaderChangeForm.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+import { Form } from 'semantic-ui-react';
+import WcaSearchFormControl from '../../SearchWidget/WcaSearchFormControl';
+import useSaveAction from '../../../lib/hooks/useSaveAction';
+import Loading from '../../Requests/Loading';
+import { COUNCILS_STATUS } from '../../../lib/helpers/groups-and-roles-constants';
+import { apiV0Urls } from '../../../lib/requests/routes.js.erb';
+
+const oldLeaderStatusOptions = ['senior_member', 'member', 'resign'];
+const oldLeaderStatusOptionsMap = {
+  senior_member: 'Senior Member',
+  member: 'Member',
+  resign: 'Resign',
+};
+
+export default function LeaderChangeForm({
+  setEditLeader, syncData, group, oldLeader,
+}) {
+  const [formValues, setFormValues] = useState({
+    newLeader: null,
+    oldLeaderStatus: oldLeaderStatusOptions[0],
+  });
+  const { save } = useSaveAction();
+  const [saving, setSaving] = useState(false);
+
+  const handleFormChange = (_, { name, value }) => setFormValues({ ...formValues, [name]: value });
+
+  const changeOldLeaderStatus = async () => new Promise((resolve) => {
+    if (!oldLeader) { // No old leader
+      resolve();
+      return;
+    }
+    if (formValues.oldLeaderStatus === oldLeaderStatusOptions[2]) { // Resign
```

Comment:
Relying on `oldLeaderStatusOptions[2]` to be the resign case seems prone to future bugs. An enum would be nice but I guess we don't really have those in js. I'd rather just use 'resign' here (maybe via a top-level `const RESIGN_KEY = 'resign'`?), since then the code is clearer and if the options are changed later on, there's no chance of another option accidentally carrying out the resign action.

Reply:
I've done something similar, hope this is fine?